apps/rotm/acceptance/features/contact-details.js

```diff
@@ -0,0 +1,119 @@
+  I,
+  contactDetailsPage
+) => {
+  I.dontSeeElements([
+    contactDetailsPage.id.email,
+    contactDetailsPage.id.phone,
+    contactDetailsPage.id.phone2,
+  ]);
+});
+
+Scenario('I see errors if I try and submit an incomplete form', (
+  I,
+  contactDetailsPage
+) => {
+  I.submitForm();
+  I.seeErrors([
+    contactDetailsPage.id.name,
+    contactDetailsPage.id.contactType
+  ]);
+});
+
+Scenario('I see an error if I submit the form without an email address', (
+  I,
+  contactDetailsPage
+) => {
+  I.fillField(contactDetailsPage.id.name, contactDetailsPage.content.name);
+  I.checkOption(contactDetailsPage.id.emailOption);
+  I.submitForm();
+  I.seeErrors(contactDetailsPage.id.email);
+});
+
+Scenario('I see an error if I submit the form with an invalid email address', (
+  I,
+  contactDetailsPage
+) => {
+  I.fillField(contactDetailsPage.id.name, contactDetailsPage.content.name);
+  I.checkOption(contactDetailsPage.id.emailOption);
+  I.fillField(contactDetailsPage.id.email, contactDetailsPage.content.invalidEmail);
+  I.submitForm();
+  I.seeErrors(contactDetailsPage.id.email);
+});
+
+Scenario('I see an error if I submit the form without a phone number (phone)', (
+  I,
+  contactDetailsPage
+) => {
+  I.fillField(contactDetailsPage.id.name, contactDetailsPage.content.name);
+  I.checkOption(contactDetailsPage.id.phoneOption);
+  I.submitForm();
+  I.seeErrors(contactDetailsPage.id.phone);
+});
+
+Scenario('I see an error if I submit the form with an invalid phone number (phone)', (
+  I,
+  contactDetailsPage
+) => {
+  I.fillField(contactDetailsPage.id.name, contactDetailsPage.content.name);
+  I.checkOption(contactDetailsPage.id.phoneOption);
+  I.fillField(contactDetailsPage.id.phone, contactDetailsPage.content.invalidPhone);
+  I.submitForm();
+  I.seeErrors(contactDetailsPage.id.phone);
+});
+
+Scenario('I see an error if I submit the form without a phone number (text)', (
+  I,
+  contactDetailsPage
+) => {
+  I.fillField(contactDetailsPage.id.name, contactDetailsPage.content.name);
+  I.checkOption(contactDetailsPage.id.textOption);
+  I.submitForm();
+  I.seeErrors(contactDetailsPage.id.phone2);
+});
+
+Scenario('I see an error if I submit the form with an invalid phone number (text)', (
+  I,
+  contactDetailsPage
+) => {
+  I.fillField(contactDetailsPage.id.name, contactDetailsPage.content.name);
+  I.checkOption(contactDetailsPage.id.textOption);
+  I.fillField(contactDetailsPage.id.phone2, contactDetailsPage.content.invalidPhone);
+  I.submitForm();
+  I.seeErrors(contactDetailsPage.id.phone2);
+});
+
+Scenario('I am taken to the confirm page on a valid submission', (
+  I,
+  contactDetailsPage,
+  confirmPage
+) => {
+  I.fillField(contactDetailsPage.id.name, contactDetailsPage.content.name);
+  I.checkOption(contactDetailsPage.id.emailOption);
+  I.fillField(contactDetailsPage.id.email, contactDetailsPage.content.validEmail);
+  I.submitForm();
+  I.seeInCurrentUrl(confirmPage.url);
+});
```

Comment:
If we get too programmatic it gets quite far from traditional acceptance testing